test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -312,14 +365,16 @@ int main() try
         ::Test::Bootstrap::CleanupBootstrap();
     });
 
-
     ::Test::Bootstrap::SetupBootstrap();
 
     PushNotificationActivationInfo info(
         PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator,
         winrt::guid(c_comServerId)); // same clsid as app manifest
 
-    g_appToken = PushNotificationManager::RegisterActivator(info);
+    if (Test::AppModel::IsPackagedProcess())
+    {
+        g_appToken = PushNotificationManager::RegisterActivator(info);
+    }
```

Comment:
Why didn't you use the IsSupported method here instead and branch the code out based on that?

Reply:
I wanted to test the IsSupported method just by itself.

Reply:
This is done via scope_exit we define at the beginning of main. This block also handles cleaning up bootstrap.

Reply:
Then in that case, the scope exit should be right below the RegisterActivator.

Reply:
Also why isn't there a Protocol activator registration here. Is Main running only for packaged apps or will this implementation be shared by both packaged and unpackaged?

Reply:
It's going to be shared by packaged/unpackaged. But we can consider splitting the test app into two separate apps.

Reply:
A lot of odd things going on here in main IMHO:
1) I feel that Registration should be defined by the test as part of LaunchArgs. Automatically doing this for every test seems a bit of an anti-pattern for testing.

2) If we are running code for both packaged and unpackaged, it stands to reason that we cover both registration flows as part of this. Same for Unregister.

Reply:
We can fix the unit tests to handle the concerns in (1). And I agree that we should cover both registration flows for pkg/unpkg, if we try to register with ProtocolActivator right now the function will just throw. This PR only contains the IsActivatorSupported function but we can definitely implement (2) when we start implementing unpkg.